runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -45,6 +46,19 @@ private Utils() {
 
     private static final char[] hexArray = "0123456789ABCDEF".toCharArray();
 
+
+    /**
+     * When true randomness is not required using UUID.randomUUID() can be really slow.
+     * Blocking for 50+ ms for entropy to build up is not unusual. This method generates
+     * random UUIDs based on a LCG with a thread local seed.
+     * @return A psuedorandom
+     */
+    public static UUID genPseudorandomUUID() {
```

Comment:
>Using UUID.randomUUID for transaction IDs can be really slow because it uses secureRandom.
```
Benchmark                 Mode         Cnt                 Score 
javaUUIDGenerator         thrpt     2_613_099           ops/s
```
UUID.randomUUID() is actually 100k times faster than 50ms

Reply:
Run the benchmark with `-Djava.security.egd=file:/dev/random` I think macos uses urandom and not random as a default. urandom is psuedorandom so it won't block. 

SecureRandom consumes bytes from  `/proc/sys/kernel/random/entropy_avail`. Also, In a real system multiple threads consume bytes from the entropy pool, so having a single thread consume the bytes sequentially is not a good benchmark.

Reply:
It's hard to repro without using the bouncy castle provider

![image](https://user-images.githubusercontent.com/7818642/92672476-9cbbeb00-f2cd-11ea-9968-1b51c70c5b4a.png)